src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/workflow/WorkflowManager.kt

```diff
@@ -0,0 +1,88 @@
+package com.cognifide.gradle.aem.common.instance.service.workflow
+
+import com.cognifide.gradle.aem.common.instance.InstanceException
+import com.cognifide.gradle.aem.common.instance.InstanceService
+import com.cognifide.gradle.aem.common.instance.InstanceSync
+import com.cognifide.gradle.aem.common.instance.service.repository.Node
+import com.cognifide.gradle.aem.common.utils.Formats
+
+class WorkflowManager(sync: InstanceSync) : InstanceService(sync) {
+
+    val repository = sync.repository
+
+    fun toggleWhile(names: List<String>, expectedState: Boolean, callback: () -> Unit) {
```

Comment:
this method should operator on single `name`, extended signature/overloaded should work on `names: Iterable<String>`

Reply:
That way we could have some duplicated code as it is not reusable easily

If we made a method for a single `string` and tried to call it inside `Iterable` method, it would call `callback` for each toggled workflow, which is not preferred.